tests/integration/sharding_tests/test_mongos.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import pytest
+from pymongo.errors import OperationFailure
+from pytest_operator.plugin import OpsTest
+
+from .helpers import count_users, generate_mongodb_client, get_username_password
+
+SHARD_ONE_APP_NAME = "shard-one"
+MONGOS_APP_NAME = "mongos-k8s"
+CONFIG_SERVER_APP_NAME = "config-server-one"
+SHARD_REL_NAME = "sharding"
+CLUSTER_REL_NAME = "cluster"
+CONFIG_SERVER_REL_NAME = "config-server"
+TIMEOUT = 10 * 60
+
+
+@pytest.mark.group(1)
+@pytest.mark.skip("Will be enabled after DPE-5040 is done")
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest, mongos_host_application_charm) -> None:
+    """Build and deploy a sharded cluster."""
+    mongodb_charm = await ops_test.build_charm(".")
+    await ops_test.model.deploy(
+        mongodb_charm,
+        num_units=1,
+        config={"role": "config-server"},
+        application_name=CONFIG_SERVER_APP_NAME,
+    )
+    await ops_test.model.deploy(
+        mongodb_charm, num_units=1, config={"role": "shard"}, application_name=SHARD_ONE_APP_NAME
+    )
+
+    await ops_test.model.deploy(
+        MONGOS_APP_NAME,
+        channel="6/edge",
+        revision=3,
```

Comment:
no revision for now since we will need to wait for it to be published

Reply:
Done in 92e9d3c